arPullGlob.py

```diff
@@ -22,108 +24,71 @@
-                             # remember to adjust CM number in PV if greater than '10'
-                else:
-                    for j in range(1, 9):
-                        pvList.append("ACCL:L" + str(lin) + "B:" + str(n) + str(j))
-                             # Finally, do the 20 CMs in L3B
-        if lin == 3:
-            for n in range(16, 36):
-                for j in range(1, 9):
-                    pvList.append("ACCL:L" + str(lin) + "B:" + str(n) + str(j))
-
+    for lo in LINAC_OBJECTS:
+        for cm in lo.cryomodules:
+            for cav in lo.cryomodules[cm].cavities:
+                pvList.append(lo.cryomodules[cm].cavities[cav].pvPrefix)
     return pvList;
 
 
-
-class Archiver(object):
-
-
-    def __init__(self, machine):
-        # type: (str) -> None
-        self.url_formatter = ARCHIVER_URL_FORMATTER.format(MACHINE=machine)
-        
-        #******************************************************
-
 #******************************************************
 
-    def getValuesOverTimeRange(self, strtTim, endTim, timeInterval=None):
+def getValuesOverTimeRange(strtTim, endTim, timeInterval=None):
  #       # type: (List[str], datetime, datetime, int) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
-        global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
-        pvList=[]
-        cavPvRfRdy={}
-        cavPvFB={}
-        cavPvIntlk={}
+    global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
+    pvList=[]
+    results={}
+    cavPvRfRdy=[]
+    cavPvFB=[]
+    cavPvIntlk=[]
+    cavPv=[]
 
-        #global cavPvFB, cavPvIntlk, cavPvRfRdy
-        if (resultsFB == {}) or (strtTim != (time.mktime(datetime.datetime.startTime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))) or (endTime != int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))):
-            startTime = (time.mktime(datetime.datetime.startTime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))
-            endTime = int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))
-            cavPv=makList()
-            codeFlt = ["0:FB_SUM", "0:RFS:INTLK_FIRST", "0:RFREADYFORBEAM"]
-            for pv in cavPv:
-                    cavPvFB.append(str(pv)+codeFlt[0])
-                    cavPvIntlk.append(str(pv)+codeFlt[1])
-                    cavPvRfRdy.append(str(pv)+codeFlt[2])
-            for pv in cavPv:
-                for i in range(3):
-                    pvList.append(str(pv) + codeFlt[i])
-            
-            url = self.url_formatter.format(SUFFIX=RANGE_RESULT_SUFFIX)
+    archiver=Archiver("lcls")
+
+    #global cavPvFB, cavPvIntlk, cavPvRfRdy
+    if (resultsFB == {}) or (strtTim != (time.mktime(datetime.datetime.strptime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))) or (endTime != int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))):
```

Comment:
This line is super long, it would help to break it up a bit for readability

Reply:
Agreed. I don't know what half of it means. He's checking the formatting of the time variables? There's gotta be a better way.